src/main/scala/beam/agentsim/agents/rideHail/RideHailingManager.scala

```diff
@@ -816,12 +821,16 @@ class RideHailingManager(
       (rideHailingAgentLocation, distance)
     })
     //TODO: Possibly get multiple taxis in this block
-    distances2RideHailingAgents.filterNot(x => lockedVehicles(x._1.vehicleId)).sortBy(_._2)
+    distances2RideHailingAgents.filterNot(x => lockedVehicles(x._1.vehicleId))
   }
 
   def getClosestIdleRideHailingAgent(pickupLocation: Coord,
                                      radius: Double): Option[(RideHailingAgentLocation, Double)] = {
-    getClosestIdleVehiclesWithinRadius(pickupLocation, radius).headOption
+    val idleVehicles = getIdleVehiclesWithinRadius(pickupLocation, radius)
+    if (idleVehicles.isEmpty) None
+    else {
+      Some(idleVehicles.minBy { case (location, radius) => radius })
```

Comment:
I assume you went with this for readability over `Some(idleVehicles.minBy(_._2))`?

Reply:
Yes, right. I think this `Some(idleVehicles.minBy(_._2))` is pretty hard to read and understand, if you're not an author (even if you're author, will forget it fast :D)